archivist/sboms.py

```diff
@@ -66,12 +69,149 @@ def __init__(self, archivist: "type_helper.Archivist"):
+
+        else:
+            url = data["url"]
+            fd = BytesIO()
+            get_url(url, fd)
+            sbom = xmltodict_parse(fd, xml_attribs=True, disable_entities=False)
+
+        b = sbom["bom"]
+        m = b["metadata"]
+        c = m["component"]
+        hash_value = c["hashes"]["hash"]["#text"]
+        result = {
+            "author": c["author"],
+            "component": c["name"],
+            "hash": hash_value,
+            "repo": "",  # unsure about this
+            "supplier": c["supplier"]["name"],
+            "uuid": b["@serialNumber"],
+            "version": c["version"],
+        }
+
+        return result
+
+    def create(self, data: Dict) -> Dict:  # pragma: no cover
+        """
+        Create an sbom and return struct suitable for use in an asset
+        or event creation.
+
+        Args:
+            data (dict): dictionary
+
+        A YAML representation of the data argument would be:
+
+            .. code-block:: yaml
+
+                filename: functests/test_resources/sboms/gen1.xml
+                content_type: text/xml
+                confirm: True
+                params:
+                  privacy: PRIVATE
+
+            OR
+
+            .. code-block:: yaml
+
+                url: https://some.hostname/cdx.xml
+                content_type: text/xml
+                confirm: True
+                params:
+                  privacy: PRIVATE
+
+             Either 'filename' or 'url' is required.
+             'content_type' is required.
+
+        Returns:
+
+            A dict suitable for adding to an asset or event creation
+
+        A YAML representation of the result would be:
+
+            .. code-block:: yaml
+
+                arc_display_name: Acme Generation1 SBOM
+                arc_attachment_identity: sboms/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+                .....
+
+        """
+        result = None
+        filename = data.get("filename")
+        if filename is not None:
+            with open(filename, "rb") as fd:
+                sbom = self.upload(
+                    fd,
+                    confirm=data.get("confirm", False),
+                    mtype=data.get("content_type"),
+                    params=data.get("params"),
+                )
+
+        else:
+            url = data["url"]
+            fd = BytesIO()
+            get_url(url, fd)
+            sbom = self.upload(
+                fd,
+                confirm=data.get("confirm", False),
+                mtype=data.get("content_type"),
+                params=data.get("params"),
+            )
+
+        # response to sbom upload contains all the info we need.
+        s = sbom.dict()
+        _, hash_value = s["hashes"][0].split(":")
+        result = {
+            "author": ",".join(s["authors"]),
+            "component": s["component"],
+            "identity": s["identity"],
+            "hash": hash_value,
+            "repo": "",  # unsure about this
```

Comment:
As prior comment, leave it blank for now.

Reply:
removed the line